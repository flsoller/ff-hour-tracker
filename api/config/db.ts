export const CONNECTION_STRING = 'mongodb://mongo:27017/hour-tracker';
