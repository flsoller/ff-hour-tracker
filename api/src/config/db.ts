import { Sequelize } from 'sequelize';

const sequelize = new Sequelize('app-db', 'postgres', 'postgres', {
  host: 'postgres',
  dialect: 'postgres',
});

export const connectDb = async () => {
  try {
    await sequelize.authenticate();
    console.log('Connection has been established successfully.');
  } catch (error) {
    console.error('Unable to connect to the database:', error);
  }
};
