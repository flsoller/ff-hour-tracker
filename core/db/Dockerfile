# build / dev stage
FROM node:20.12.2-alpine AS build
WORKDIR /app

COPY package.json .
COPY yarn.lock .
COPY .yarnrc.yml .
COPY .yarn/releases .yarn/releases/
COPY core/db/package.json ./core/db/package.json

RUN yarn install

COPY core/db ./core/db

RUN yarn core:db:build

COPY core/db/src/migrations ./core/db/dist/migrations/

EXPOSE 5420

ENTRYPOINT ["tail", "-f", "/dev/null"]