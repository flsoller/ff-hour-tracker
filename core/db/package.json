{
  "name": "@hour-tracker/core-db",
  "description": "Database package for ff-hour-tracker app",
  "license": "UNLICENSED",
  "main": "./dist/index.js",
  "scripts": {
    "test": "dotenv -e .env.test jest -i --",
    "build": "esbuild --outdir=dist src/*/*.ts src/*.ts --bundle --minify --sourcemap --platform=node --target=es2020",
    "db:seed:dev": "dotenv -e .env.dev tsx ./src/seeder/seed.ts",
    "db:migration:dev:generate": "sh -c 'drizzle-kit generate --name ${0}'",
    "db:migration:dev:migrate": "dotenv -e .env.dev drizzle-kit migrate",
    "db:migration:test:migrate": "dotenv -e .env.test drizzle-kit migrate",
    "db:migrate": "node ./dist/migrate.js"
  },
  "dependencies": {
    "@neondatabase/serverless": "0.9.3",
    "drizzle-orm": "0.31.0",
    "postgres": "3.4.4"
  },
  "devDependencies": {
    "@types/jest": "29.5.12",
    "dotenv": "16.4.5",
    "dotenv-cli": "7.4.2",
    "drizzle-kit": "0.22.1",
    "esbuild": "0.21.4",
    "jest": "29.7.0",
    "ts-jest": "29.1.4",
    "tsx": "4.11.0",
    "typescript": "5.4.5"
  }
}
