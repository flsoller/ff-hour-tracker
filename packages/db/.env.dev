DATABASE_URL="postgres://postgres:postgres@localhost:5432/app_db_dev"
DEV_OR_INTEGRATION_USE="true"
ADMIN_USER_EMAIL="admin@user.com"
ADMIN_USER_PW="support-user-pw"