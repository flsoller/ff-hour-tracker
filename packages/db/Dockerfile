# build / dev stage
FROM node:20.12.2-alpine AS build
WORKDIR /app

COPY package.json .
COPY yarn.lock .
COPY .yarnrc.yml .
COPY .yarn/releases .yarn/releases/
COPY turbo.json ./
COPY packages/db/package.json ./packages/db/package.json

RUN yarn install

COPY packages/db ./packages/db

RUN yarn turbo:build

COPY packages/db/src/migrations ./packages/db/dist/migrations/

EXPOSE 5420

ENTRYPOINT ["tail", "-f", "/dev/null"]