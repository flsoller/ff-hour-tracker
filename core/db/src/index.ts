import { localDb } from './clients/pg';
import { serverlessDb } from './clients/serverless-pg';
import { USE_LOCAL_CLIENT } from './config';

// Model imports for exposing them with the db package
import { activityTypes } from './schema/activity-type';
import { members } from './schema/member';
import { organizations } from './schema/organization';
import { timeLogs } from './schema/time-log';
import { users } from './schema/user';

/**
 * Gets the necessary database client depending on where this package is used.
 * Neon serverless driver communicates over http which can not be used for local
 * and integration test setups.
 */
function getDbClient() {
  return USE_LOCAL_CLIENT ? localDb : serverlessDb;
}

export const db = getDbClient();
export const models = {
  activityTypes,
  members,
  organizations,
  timeLogs,
  users,
};
