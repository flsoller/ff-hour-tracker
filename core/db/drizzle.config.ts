import { defineConfig } from 'drizzle-kit';
import { CONNECTION_STRING } from './src/config';

export default defineConfig({
  dialect: 'postgresql',
  schema: './src/schema/*',
  out: './src/migrations',
  dbCredentials: {
    url: CONNECTION_STRING,
  },
  migrations: {
    table: 'migrations',
    schema: 'hour_tracker_migrations',
  },
  breakpoints: false,
  schemaFilter: 'hour_tracker',
});
