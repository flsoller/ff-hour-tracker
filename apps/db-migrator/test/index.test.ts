import { sql } from "drizzle-orm";
import { db } from "@hour-tracker/db";
import postgres from "postgres";

describe("Migrations", () => {
  it("should apply all migrations to provided DB", async () => {
    const migrations = (await db.execute(
      sql`SELECT * FROM hour_tracker_migrations.migrations`
    )) as postgres.RowList<Record<string, unknown>[]>;
    expect(migrations).toHaveLength(1);
    migrations.map((migration) => {
      expect(migration).toMatchSnapshot({
        created_at: expect.any(String),
      });
    });
  });
});
