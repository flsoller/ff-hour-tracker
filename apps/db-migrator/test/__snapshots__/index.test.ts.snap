// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Migrations should apply all migrations to provided DB 1`] = `
{
  "created_at": Any<String>,
  "hash": "b1b46e7d016aa5eadbb00c35014b3303d3972618801b8304c187f5fdbf7393c4",
  "id": 1,
}
`;

exports[`Migrations should apply all migrations to provided DB 2`] = `
{
  "created_at": Any<String>,
  "hash": "94ffbd5783d22db08782fbcf7cc3285e3217a0776f3eabcaf62c62d9d325d626",
  "id": 2,
}
`;
